src/backend/gporca/libgpopt/src/translate/CTranslatorExprToDXL.cpp

```diff
@@ -1262,8 +1255,7 @@ CTranslatorExprToDXL::PdxlnDynamicBitmapTableScan(
 
 	// build projection list
 	CColRefSet *pcrsOutput = pexprScan->Prpp()->PcrsRequired();
-	AddBitmapFilterColumns(m_mp, pop, pexprRecheckCond, pexprScalar,
-						   pcrsOutput);
+	AddBitmapFilterColumns(m_mp, pop, pexprScalar, pcrsOutput);
```

Comment:
I've tried to validate an original commit https://github.com/arenadata/gpdb/commit/97edf9b1cbd47c33918e5780e04ea241b59a2f46 that implements this behavior. I found that this patch doesn't affect any `DynamicBitmapHeapScan` tests. So I tried to find out why. There is an appropriate [test case](https://github.com/arenadata/gpdb/blob/adb-6.x/src/backend/gporca/data/dxl/minidump/sql/DynamicBitmapIndexScan.sql) in the original commit. This test case helped me after small corrections (including changing value of `allow_system_table_mods` and shrinking namespace filters) to generate a plan with an excessive column `flex_value_id` in the target list:

```
test=# explain (costs off, verbose) select inner_table.flex_value_set_id from inner_table where inner_table.flex_value_id = 1;
                                      QUERY PLAN                                       
---------------------------------------------------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3)
   Output: flex_value_set_id
   ->  Result
         Output: flex_value_set_id
         ->  Sequence
               Output: flex_value_set_id, flex_value_id
               ->  Partition Selector for inner_table (dynamic scan id: 1)
                     Partitions selected: 6 (out of 6)
               ->  Dynamic Bitmap Heap Scan on public.inner_table (dynamic scan id: 1)
                     Output: flex_value_set_id, flex_value_id
                     Recheck Cond: (inner_table.flex_value_id = 1::numeric)
                     ->  Dynamic Bitmap Index Scan on inner_table_idx
                           Index Cond: (inner_table.flex_value_id = 1::numeric)
 Optimizer: Pivotal Optimizer (GPORCA)
```

But when I tried to combine Dynamic Scan with Shared Scan, I got a valid plan with projection node before materialization:

```
test=# explain (costs off, verbose) with mat as(select inner_table.flex_value_set_id i from inner_table where inner_table.flex_value_id = 1) select m1.i from mat m1 join mat m2 on m1.i = m2.i;
                                               QUERY PLAN                                                
---------------------------------------------------------------------------------------------------------
 Gather Motion 3:1  (slice2; segments: 3)
   Output: share0_ref3.flex_value_set_id
   ->  Sequence
         Output: share0_ref3.flex_value_set_id
         ->  Shared Scan (share slice:id 2:0)
               Output: share0_ref1.flex_value_set_id
               ->  Materialize
                     Output: inner_table.flex_value_set_id
                     ->  Result
                           Output: inner_table.flex_value_set_id
                           ->  Sequence
                                 Output: inner_table.flex_value_set_id, inner_table.flex_value_id
                                 ->  Partition Selector for inner_table (dynamic scan id: 1)
                                       Partitions selected: 6 (out of 6)
                                 ->  Dynamic Bitmap Heap Scan on public.inner_table (dynamic scan id: 1)
                                       Output: inner_table.flex_value_set_id, inner_table.flex_value_id
                                       Recheck Cond: (inner_table.flex_value_id = 1::numeric)
                                       ->  Dynamic Bitmap Index Scan on inner_table_idx
                                             Index Cond: (inner_table.flex_value_id = 1::numeric)
         ->  Nested Loop
               Output: share0_ref3.flex_value_set_id
...
```